### ISSUE TITLE
Cooling fan does not come on even when CPU is stressed / loaded in Cinebench.

### ISSUE THREAD

**Issue by Chatbox1024** [gnodipac886/MatebookXPro-hackintosh#84]
First of all: Absolutely fantastic work!  My Matebook X Pro (2018 i7) is running Catalina 10.15.3 really smoothly.  Thank you so much!  I love your work, and really appreciate it!

Bug details:
"MatebookXPro-hackintosh" Release: 1.1.20.2
Hardware: Matebook X Pro 2018 i7 (with KXG50ZNV512G TOSHIBA m.2 SSD), boot with external USB SSD.

Issue:
The cooling fan doesn't seem to come on even when I was trying to stress test the CPU in Cinebench.  As such, the CPU performance is lower than expected when it is under load.

**Comment by Chatbox1024** [comment]
p.s. By the way, where was AppleALC-m8 v1.4.5 compiled from?  It seems to be different than the AppleALC v1.4.5 release.

**Comment by kvn1351** [comment]
> p.s. By the way, where was AppleALC-m8 v1.4.5 compiled from? It seems to be different than the AppleALC v1.4.5 release.

It's not 1.4.5. And since 1.4.4 the patches made for our Matebook have been added to the main repo. So there's no more need for the m8 version. Just get the latest AppleALC and you're good to go.

**Comment by Chatbox1024** [comment]
> > p.s. By the way, where was AppleALC-m8 v1.4.5 compiled from? It seems to be different than the AppleALC v1.4.5 release.
> 
> It's not 1.4.5. And since 1.4.4 the patches made for our Matebook have been added to the main repo. So there's no more need for the m8 version. Just get the latest AppleALC and you're good to go.

Thank you!!  I'll do that for AppleALC.

**Comment by Chatbox1024** [comment]
I got Intel Power Gadget installed.  It looks like the laptop is CPU throttling even at 60c...and so that's why the fan doesn't even need to come on.

I wonder where this 60c temperature ceiling is set / specified / configured.

**Comment by Chatbox1024** [comment]
Found out what the issue was (mostly me, user error):
1. I had to generate the CPUFriendDataProvider.kext as documented in this repo.
2. Using the voltageshift from here: https://github.com/buliaoyin/VoltageShift
3. Set the undervolt: ./voltageshift offset -80 -50 -80
4. Set the powerlimit: ./voltageshift powerlimit 15 64 44 2
5. Set the MSR_TURBO_RATIO_LIMIT to the following:
./voltageshift write 0x1AD 0x28282828


MSR_TURBO_RATIO_LIMIT............(0x1AD) : 0x28282828
------------------------------------------
 - Maximum Ratio Limit for C01.......... : 28 (4000 MHz) 
 - Maximum Ratio Limit for C02.......... : 28 (4000 MHz) 
 - Maximum Ratio Limit for C03.......... : 28 (4000 MHz) 
 - Maximum Ratio Limit for C04.......... : 28 (4000 MHz)


This then gives me an overall initial burst of 125GFLOPS in LINPACK (for a problem size of 20,000).

**Comment by jonescamilla** [comment]
Chatbox1024 What are you referring to when you list 
" 1. I had to generate the CPUFriendDataProvider.kext as documented in this repo."
I'm lost when it comes to this.

**Comment by Chatbox1024** [comment]
Under here: https://github.com/gnodipac886/MatebookXPro-hackintosh#412019-10144--new-power-management-kexts

Run the "sh...." command from the terminal.  It'll download CPUFriend, as well as creating the CPUFriendDataProvider.kext specific to your processor.  The script will ask you what low/idle frequency would you like the CPU to run at (when it's idle, I picked 800MHz), as well as the performance preference (I picked balanced performance).

Save the two kexts to your CLOVER's kext folder (and overwrite the old/existing ones)

**Comment by jonescamilla** [comment]
At what temperature does your fan tend to kick in?

**Comment by Chatbox1024** [comment]
Around 68-70c.

**Comment by jonescamilla** [comment]
That's kind of late no?

**Comment by Chatbox1024** [comment]
For ultrabook design...no.

**Comment by Chatbox1024** [comment]
This is probably the final update for this power management matter.

The following is a screenshot of the Cinebench R20 benchmark, and LINPACK (problem size 20,00) showing the thermal-performance stability over 18 tests (around 20 minutes, full load).  On battery only.

![Screen Shot 2020-03-18 at 5 19 12 PM](https://user-images.githubusercontent.com/703217/77019677-86485e80-693e-11ea-8dfd-e6bedab0623f.png)

**Comment by jonescamilla** [comment]
Impressive numbers. My last test in Cinebench only got as high as ~1000. I've got a ways to go.

**Comment by PLTorrent** [comment]
I also saw a 60deg cap on my system. Used only this: 

Set the powerlimit: ./voltageshift powerlimit 25 64 44 2

and I got the same result as in Windows 10. 

<img width="353" alt="Screenshot 2020-05-01 at 01 05 01" src="https://user-images.githubusercontent.com/33349763/80767599-8ac08380-8b48-11ea-84b8-d6d5bbb0bbc9.png">

Thanks a lot for this tip. Curious thing though is that my CPU reports 15W TDP during test, not 25W.

**Comment by mircoianese** [comment]
> Under here: https://github.com/gnodipac886/MatebookXPro-hackintosh#412019-10144--new-power-management-kexts
> 
> Run the "sh...." command from the terminal. It'll download CPUFriend, as well as creating the CPUFriendDataProvider.kext specific to your processor. The script will ask you what low/idle frequency would you like the CPU to run at (when it's idle, I picked 800MHz), as well as the performance preference (I picked balanced performance).
> 
> Save the two kexts to your CLOVER's kext folder (and overwrite the old/existing ones)

Hello! I've done this but still getting around 70GFlops in linpack, so something must be wrong (i7 2018 MBX). CPU does not go above 60°C,  so some throttling is going on here... The fan also does not kick in full-speed. 

Any advice? Thanks

**Comment by PLTorrent** [comment]
> 
> Any advice? Thanks
> 

In my case CPUFriend did not help. What allowed me to get over the limit was this particular version of voltageshift that is mentioned in this thread. You will have to compile it yourself from github repo (the links leading to pre-built one lead to a different version, and not working anyway due to some reason) using Xcode. Just get rid of signatures and build it for local usage.

Set the powerlimit: ./voltageshift powerlimit 25 64 44 2

After this you should see improvement. Then you can try undervolting for better performance/battery life.

**Comment by mircoianese** [comment]
> ./voltageshift powerlimit 25 64 44 2

I've compiled the kext and placed it in Clover. I have removed CPUFriend kexts. After reboot I tried `./voltageshift info` and some stats were shown. I then tried to apply the powerlimit, but after that with `./voltageshift info` no more stats are shown, and the patch does not seem to work.

<img width="572" alt="screen" src="https://user-images.githubusercontent.com/22716366/80907818-497fcd80-8d1a-11ea-8c55-2b9b3c76e4a5.png">
